src/python/pants/engine/internals/scheduler_test.py

```diff
@@ -126,7 +126,12 @@ def test_strict_equals() -> None:
 # -----------------------------------------------------------------------------------------------
 
 
-@union
+@dataclass(frozen=True)
+class Fuel:
+    pass
+
+
+@union(in_scope_types=[Fuel])
```

Comment:
`in_scope` sounds weird to me. It doesn't quite give that "you have to declare additional arguments" vibe.

Reply:
> It doesn't quite give that "you have to declare additional arguments" vibe.

To be clear: you don't have to declare any additional arguments. These are just types that are guaranteed to be available to the callee. So in this example, you can see that `motorcycle_num_wheels` doesn't use `Fuel`. It could, but it doesn't need to.

Reply:
I'm open to changing the name as part of #16721 (and beyond), so I'm going to roll with this for now.